referenceframe/frame.go

```diff
@@ -400,20 +399,88 @@ func (rf *rotationalFrame) Name() string {
 }
 
 func (rf *rotationalFrame) MarshalJSON() ([]byte, error) {
-	m := FrameMapConfig{
-		"type":    "rotational",
-		"name":    rf.name,
-		"rotAxis": rf.rotAxis,
-		"limit":   rf.limits,
-	}
-	return json.Marshal(m)
+	return json.Marshal(rf.toConfig())
+}
+
+func (rf *rotationalFrame) toConfig() FrameMapConfig {
+	config := rf.baseFrame.toConfig()
+	config["type"] = "rotational"
+	config["rotAxis"] = rf.rotAxis
+	return config
 }
 
 func (rf *rotationalFrame) AlmostEquals(otherFrame Frame) bool {
 	other, ok := otherFrame.(*rotationalFrame)
 	return ok && rf.baseFrame.AlmostEquals(other.baseFrame) && spatial.R3VectorAlmostEqual(rf.rotAxis, other.rotAxis, 1e-8)
 }
 
+type linearlyActuatedRotationalFrame struct {
+	*rotationalFrame
+	a, b float64
+}
+
+// NewLinearlyActuatedRotationalFrame creates a frame that represents a robot mechanism where a linear actuator is used to achieve
+// rotational motion about a passive hinge.  This configuration forms a triangle (visualized below) where the linear actuator length is
+// side c, and the offset along the robot links from the rotation is given by a and b, with a being the length along the link previous in
+// the kinematic chain. The transformation given by this frame maps c to theta, which is the angle between a and b
+//
+//                 |
+//                 |
+//                 *
+//                /|
+//            c  / |
+//              /  | b
+//      _______/___|
+//               a
+//
+func NewLinearlyActuatedRotationalFrame(name string, axis spatial.R4AA, limit Limit, a, b float64) (Frame, error) {
+	if a <= 0 || b <= 0 {
+		return nil, errors.New("cannot create a linearlyActuatedRotationalFrame with values a || b <= 0")
+	}
+	rf, err := NewRotationalFrame(name, axis, limit)
+	if err != nil {
+		return nil, err
+	}
+	return &linearlyActuatedRotationalFrame{rotationalFrame: rf.(*rotationalFrame), a: a, b: b}, nil
+}
+
+func (larf *linearlyActuatedRotationalFrame) Transform(input []Input) (spatial.Pose, error) {
+	err := larf.validInputs(input)
+	if err != nil && !strings.Contains(err.Error(), OOBErrString) {
+		return nil, err
+	}
+
+	// law of cosines to determine theta from a, b and c
+	cosTheta := (larf.a*larf.a + larf.b*larf.b - input[0].Value*input[0].Value) / (2 * larf.a * larf.b)
+	if cosTheta < -1 || cosTheta > 1 {
+		return nil, errors.Errorf("could not transform linearly actuated rotational frame with input %f", input[0].Value)
+	}
+	return spatial.NewPoseFromOrientation(
+		r3.Vector{},
+		&spatial.R4AA{Theta: math.Acos(cosTheta), RX: larf.rotAxis.X, RY: larf.rotAxis.Y, RZ: larf.rotAxis.Z},
```

Comment:
Since this is a triangle, we'd need the less than pi `Acos` returned. golang does this automatically I think, but please confirm.